### PR TITLE
London-9-Turing-Farnoosh_Moayeri-Full-Stack-Project-Assesment

### PR DESCRIPTION
**Pull request by Farnooshmo**

### CODE REVIEW

File: client/src/App.js [hunk 1]
```diff
@@ -1,13 +1,119 @@
...
+          },
+        }
+      );
+      setVideos((prevVideos) =>
+        prevVideos.map((video) =>
+          video.id === videoId ? { ...video, rating: video.rating - 1 } : video
+        )
+      );
+    } catch (err) {
+      console.log("Error downvoting video:", err);
```
Comment (sampennington) [inline client/src/App.js:83]: Nice handling errors!

File: client/src/App.js [hunk 2]
```diff
@@ -1,13 +1,119 @@
...
+const App = () => {
+  const [videos, setVideos] = useState([]);
+
+  useEffect(() => {
+    fetchVideos();
+  }, []);
+
+  const fetchVideos = async () => {
+    try {
+      const res = await fetch("https://video-server-tumg.onrender.com/videos");
```
Comment (sampennington) [inline client/src/App.js:18]: Nice use of async/await 😄

File: server/server.js [hunk 3]
```diff
@@ -1,15 +1,125 @@
...
+  }
+  // const newVideos = {
+  //   id: idCounter,
+  //   title: req.body.title,
+  //   url: req.body.url,
+  //   rating: 0,
+  // };
+  // videos.push(newVideos);
+  // console.log("New video added:", newVideos);
+  // res.json({ id: newVideos.id });
```
Comment (sampennington) [inline server/server.js:54]: Theres lots of commented code around, would be good to clean this up, it makes it harer to review 😄

**Review by sampennington — COMMENTED** [review]
well done Farnoosh!

File: server/server.js [hunk 4]
```diff
@@ -1,15 +1,125 @@
...
+
+// Get the video with the ID
+
+app.get("/videos/:id", (req, res) => {
+  let id = parseInt(req.params.id);
+  let foundVideo = videos.find((v) => v.id === id);
+  if (foundVideo) {
+    res.json(foundVideo);
+  } else {
+    res.status(404).json({ result: "failure", message: "Video not found" });
```
[the remark this answers is not in the document]
Comment (Farnooshmo) [reply server/server.js:56]: Hi @sampennington thank you for reviewing my code. I deleted the commented lines and I also add the error.

**Pull request closed without merging**